tools/boost_names.py

```diff
@@ -43,10 +43,10 @@
 class BoostLibrary():
     def __init__(self, name: str, shared: T.List[str], static: T.List[str], single: T.List[str], multi: T.List[str]):
         self.name = name
-        self.shared = shared
-        self.static = static
-        self.single = single
-        self.multi = multi
+        self.shared = sorted(set(shared))
+        self.static = sorted(set(static))
+        self.single = sorted(set(single))
+        self.multi = sorted(set(multi))
```

Comment:
sorted returns an iterator, not a list. you need to wrap this in list as well to avoid changing the type (and running into strange errors if you try to use these more than once.

Reply:
Actually, `sorted` does return a list: https://docs.python.org/3/library/functions.html#sorted. You can verify this with `type(sorted([3, 2, 1])).__name__`.

Reply:
interesting, I could have sworn that it returned and interable. Maybe it did in python2? :shrug: